File: Trinity_code.pl
#perl Trinity.pl Fastq_files_directory

#perl code 
use strict;
use diagnostics;

#pwd should be the one where the files are downloaded; Open command prompt in that directory and use the following command to run the script
#perl RNASeq.pl /media/DSRG4new/3D or any other path in which your files are stored

my $dir = $ARGV[0];
my $file;
my $file1;
my $file2;
my @readFiles;
my $count;
my $outFile;


opendir(DIR, $dir) or die $!;
    while ($file = readdir(DIR)) {
        # Use a regular expression to ignore files beginning with a period
        #next if ($file =~ m/(^\.|_R2.fastq$|out$|bam$|gz$|gz.1$|pl$|sh$)/);
	next if ( -d $file );
	if($file =~ /(.*)\_R1\.fastq$/){
		print "$file\n";		
		@readFiles = split(/\_R1/,$file);
		$file1 = $dir."/".$readFiles[0]."_R1.fastq";
		print "$file1\n";
		$file2 = $dir."/".$readFiles[0]."_R2.fastq";
		print "$file2\n";
		$outFile = $readFiles[0]."OUT";
		print "$readFiles[0]\n";
                `Trinity --seqType fq --normalize_by_read_set --left $file1.fastq --right $file2.fastq --trimmomatic --full_cleanup --CPU 30 --max_memory 50G --bflyCPU 10 --bflyHeapSpaceMax 4G --output Trinity.$file --monitoring --verbose`;		
		$count++; 
	}
    }	
#

File: trinity_code.py
#!/usr/bin/python
import os, sys, getopt, re, subprocess

#python trinity_code.py /path/to/fastq/dir
def usage():
    print("""
trinity_code.py :   Passes paired end FASTQ files to Trinity for de novo assembly of RNA transcripts. 

trinity_code.py [-h] [-s <single end>] </path/to/directory>

    -h              Print this message

    -s              If single end reads, pass only one file to Trinity

    <directory>     Directory containg reads in FASTQ format
""")    

# Define shell arguments
opt, req = getopt.getopt(sys.argv[1:], 'hs')
opts = {}
for k,v in opt:
    opts[k] = v
if '-h' in opts.keys():
    usage(); sys.exit()
if '-s' in opts.keys():
    pass

dir_path = sys.argv[1]

# Ensure non-empty directory
if not os.listdir(dir_path):
    sys.exit("Directory %s is empty." %dir_path)

contents = os.listdir(dir_path)
r1_pattern = re.compile('(.*)_R1.fastq$') # search for fastq files ending in '_R1'
for i in range(len(contents)):
    if r1_pattern.match(contents[i]):
        id = contents[i].split('_R1')
        if id[0]+'_R2.fastq' in contents: # ensures only paired reads
            print(id[0])
            file1 = dir_path + "/" + id[0] + "_R1.fastq"
            print(file1)
            file2 = dir_path + "/" + id[0] + "_R2.fastq"
            print(file2)
            os.system(f"Trinity --seqType fq --normalize_by_read_set --left {file1} --right {file2} --trimmomatic --full_cleanup --CPU 30 --max_memory 50G --bflyCPU 10 --bflyHeapSpaceMax 4G --output {dir_path}/Trinity.{id[0]} --monitoring --verbose")
            

    